### ISSUE TITLE
Cache signed asset URLs in Cloudflare Workers KV Storage

### ISSUE THREAD

**Issue by arunsathiya** [arunsathiya/portfolio#76]
I think there is scope is improving performance by caching signed asset URLs in Cloudflare Workers KV Storage and reading from it, rather than having to generate a new signed URL on each visit by every visitor.

**Comment by arunsathiya** [comment]
For example, images in a blog post seem to take ~770ms each, could shave off a lot of that processing time by caching on Workers KV Storage:

<img width="1700" alt="image" src="https://github.com/user-attachments/assets/5db246fa-7b02-4982-8b18-9707104ac1d8">

**Comment by arunsathiya** [comment]
Implemented in 226d5418468b604388cbe09f5c1b0428e2508c49. I think it works:

```sh
GET https://www.arun.blog/cdn/blog/assets/private-among-us-games-on-the-tailscale-network-ad192dfe-ca42-414e-958c-c99ceaf84bdf.png - Ok @ 9/1/2024, 9:19:59 PM
  (log) Key: blog/assets/private-among-us-games-on-the-tailscale-network-ad192dfe-ca42-414e-958c-c99ceaf84bdf.png
  (log) Used cached signed URL for blog/assets/private-among-us-games-on-the-tailscale-network-ad192dfe-ca42-414e-958c-c99ceaf84bdf.png
```
